src/react/useQueries.ts

```diff
@@ -3,31 +3,114 @@ import React from 'react'
 import { notifyManager } from '../core/notifyManager'
 import { QueriesObserver } from '../core/queriesObserver'
 import { useQueryClient } from './QueryClientProvider'
+import { useQueryErrorResetBoundary } from './QueryErrorResetBoundary'
 import { UseQueryOptions, UseQueryResult } from './types'
 
-export function useQueries(queries: UseQueryOptions[]): UseQueryResult[] {
+export function useQueries<TData = unknown, TError = unknown>(
```

Comment:
Nice work!

I have a question. This adds the ability to specify generics for `useQueries`. Reading this I think that this may fall into the ["generic that is only used in the return position" anti-pattern](https://github.com/microsoft/TypeScript/issues/36062#issuecomment-571799283) category.

It's possible I'm reading it wrong so I wanted to ask a clarifying question: Does the approach here allow a user to specify types using the generic parameters that would differ from the return type of the query or of the `select` function?

If that's possible then the generics would not be adding a great deal of value as incorrectly typed data could still flow through.

Is that possible in this case? 

[As discussed](https://github.com/tannerlinsley/react-query/pull/1527#issuecomment-761564193), I'm generally keen on making use of type inference. But in the absence of that, avoiding type definitions that can possibly let incorrect types through is still worth aiming for.